pkg/controller/direct/firestore/firestoredatabase_controller.go

```diff
@@ -237,12 +237,18 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
 
 	updateMask := &fieldmaskpb.FieldMask{}
 	if !reflect.DeepEqual(resource.ConcurrencyMode, a.actual.ConcurrencyMode) {
-		newDb.ConcurrencyMode = resource.ConcurrencyMode
-		updateMask.Paths = append(updateMask.Paths, "concurrency_mode")
+		// Skip update if concurrency_mode is unspecified
+		if resource.ConcurrencyMode != firestorepb.Database_CONCURRENCY_MODE_UNSPECIFIED {
```

Comment:
If i read the code correctly, this is the `spec.concurrencyMode` field. What does `spec.concurrencyMode == Database_CONCURRENCY_MODE_UNSPECIFIED` mean? Is this the default value if user not specifying the spec.concurrencyMode? Can we add a test suite (maybe scenario test) for this case?

Reply:
You are right, `Database_CONCURRENCY_MODE_UNSPECIFIED` is the value that gets assigned by `FirestoreDatabaseSpec_ToProto` when `spec.concurrencyMode` is not specified. However, GCP assigns some other default value (`Database_PESSIMISTIC`). And, potentially, the field could be managed separately from KCC, assigning some other value. This case is already covered by the `firestoredatabase-minimal` test.

Reply:
I see. That makes sense. So basically this is the default handling